src/app.rs

```diff
@@ -2350,18 +2342,51 @@ impl Application for App {
                                 };
                             }
                             if !found_desktop_exec {
-                                match open::that_detached(&path) {
-                                    Ok(()) => {
-                                        let _ = recently_used_xbel::update_recently_used(
-                                            &path,
-                                            App::APP_ID.to_string(),
-                                            "cosmic-files".to_string(),
-                                            None,
-                                        );
-                                    }
-                                    Err(err) => {
-                                        log::warn!("failed to open {:?}: {}", path, err);
+                                let file_extension = path.extension();
+                                match file_extension {
+                                    Some(ext) if ext == OsStr::new("AppImage") => {
+                                        // Set the executable permission to the file
+                                        let mut perms = fs::metadata(&path)
+                                            .expect("Failed to get metadata")
+                                            .permissions();
+                                        // Set the executable permission to the file
+                                        perms.set_mode(0o755);
```

Comment:
This should involve a dialog to ask to mark it as executable. Then the bits for executable should be ORed on to the current mode, rather than indiscriminately setting the mode to 755.